packages/atlas/src/api/hooks/nfts.ts

```diff
@@ -11,35 +11,33 @@ import {
 } from '@/api/queries'
 import { createLookup } from '@/utils/data'
 
-export type NftStatus =
+type CommonNftProperties = {
+  title: string | null | undefined
+  duration: number | null | undefined
+  views: number | undefined
+}
+
+export type NftStatus = (
   | {
       status: 'auction'
-      type: 'open-auction' | 'english-auction'
+      type: 'open' | 'english'
       startingPrice: number
-      buyNowPrice?: number
-      topBid?: number
-      topBidder?: BasicMembershipFieldsFragment
-      title?: string | null
-      duration?: number | null
-      views?: number
+      buyNowPrice: number | undefined
+      topBid: number | undefined
+      topBidder: BasicMembershipFieldsFragment | undefined
```

Comment:
I'm kinda don't understand this. Why did you get rid of question mark here and replaced it with `undefined`?  I think I  don't understand your intentions here - could you explain?

Reply:
The reasoning behind this is that I want these props to `always` be passed from the `useNftWidget` hook, they can be undefined that's fine, but since they are so many it's easy to forget to add a variable in there and the compiler wouldn't complain if we marked them as optional.

So in this case, if you forget to pass a variable the compiler will yell at you, so it's a way of preventing future bugs in this component where we don't pass props manually

Reply:
hmm okay, just tried to mess around with this hook, and now makes more sense. Thank you for the explanation!